# 1. előadás

_TODO: Bevezetés, technikai tudnivalók_

 * print függvény
 * alapvető aritmetikai műveletek
 * int, float és string típusok
 * megjegyzés

## Tananyag

Az elméleti anyag a tantárgy hivatalos [Github](https://github.com/sze-info/GKNB_INTM112) felületén érhető el. A félévi anyagok folyamatosan kerülnek kidolgozásra és feltöltésre a repozitóriumba. A gyakorlati feladatok a a tantárgy [Szelearning](https://szelearning.sze.hu/course/view.php?id=13977) felületén is elérhetőek lesznek. Ez jelenleg kidolgozás alatt van.

A Github felületen elérhető tananyag online is böngészhető, de [Git](https://github.com/git-guides) segítségével le is lehet tölteni és offline használni.

Ehhez klónozni szükséges a repozitóriumot:

```git clone https://github.com/sze-info/GKNB_INTM112.git```

Utána a klónozás helyén az elérhetővé válik offline módon. Ha időközben a tananyagban frissítés történt, akkor a következő, az offline repozitórium helyén kiadott paranccsal lehet az offline anyagot frissíteni:

```git pull```

[További információ a Git telepítéséről.](https://github.com/git-guides/install-git)

[Github cheatsheet (hasznos parancsok)](https://education.github.com/git-cheat-sheet-education.pdf)

A Github-on található anyagok ```ipynb``` kiterjesztésűek. Offline megtekintésre jó választás a [Jupyter](https://jupyter.org/).

A tantárgyhoz tankönyv nem tartozik, de javasoljuk [Eric Matthes „Python Crash Course”](https://nostarch.com/python-crash-course-3rd-edition) c. könyvét, valamint a könyvhöz tartozó [Github repozitóriumot](https://github.com/ehmatthes/pcc).

## Tantárgy teljesítése

A tantárgy teljesítéséhez az óralátogatás nem szükséges, katalógus nincs. A félév során zárthelyi dolgozat nincs, az aláírás automatikusan jár minden résztvevőnek.

A félév gyakorlati vizsgával zárul a vizsgaidőszakban. A vizsga során 5 darab gyakorlati feladatot kell megoldani. A helyesen megoldott feladatok száma meghatározza a vizsga érdemjegyét.

A vizsgafeladatok megoldása a Szelearning felületen fog történni, automatizált azonnali kiértékeléssel. Ehhez mintafeladatokat a félév során fogunk nézni, jelenleg kidolgozás alatt van a Szelearning felület.

## Helló, Világ!

Első programunk a "Helló, Világ!" üzenetet jelenti meg a képenyőn. [Régi szokás szerint a legtöbb tankönyv, tananyag ezzel az egyszerű programmal indít.](https://en.wikipedia.org/wiki/%22Hello,_World!%22_program)



In [1]:
print("Helló, Világ!")

Helló, Világ!


A programunk a ```print``` kulcsszóval kezdődik, ami egy függvénynek a neve. A függvények kerek zárójelpár között várnak meghatározott paramétereket. A ```print``` függvény számára azt kell megadni, hogy mit írjon a kimenetre.

[Bővebben a ```print``` függvényről.](https://www.w3schools.com/python/ref_func_print.asp)

Bármilyen tetszőleges szöveget megadhatunk idézőjelek (```"```) vagy aposztrófok (```'```) között. Mindegy, hogy melyik használjuk, nincs különbség.

In [2]:
print("Rendőr, postás, pék is lennék, / kertésznek is vígan mennék, / de leginkább azért főleg / machine learning szakértőnek.")

Rendőr, postás, pék is lennék, / kertésznek is vígan mennék, / de leginkább azért főleg / machine learning szakértőnek.


In [3]:
print('Rendőr, postás, pék is lennék, / kertésznek is vígan mennék, / de leginkább azért főleg / machine learning szakértőnek.')

Rendőr, postás, pék is lennék, / kertésznek is vígan mennék, / de leginkább azért főleg / machine learning szakértőnek.


Természetesen egy program nem csak egy sorból állhat. Lehetőségünk van arra is, hogy egymás után több utasítást adjunk meg. Például több ```print``` függvényt írunk egymás utáni sorokba.

Ilyenkor az egyes utasítások egymás után, fentről lefelé haladva kerülnek végrehajtása. Ezt nevezzük szekvenciális végrehajtásnak. A kimeneten is ennek megfelelően jelennek meg egymás után (egymás alatt) az üzenetek.

Így [Varró Dániel](https://hu.wikipedia.org/wiki/Varr%C3%B3_D%C3%A1niel) versét törtvonalak nélkül, több sorban is megjeleníthetjük:

In [4]:
print("Rendőr, postás, pék is lennék,")
print("kertésznek is vígan mennék,")
print("de leginkább azért főleg")
print("machine learning szakértőnek.")

Rendőr, postás, pék is lennék,
kertésznek is vígan mennék,
de leginkább azért főleg
machine learning szakértőnek.


## Aritmetikai műveletek


A ```print``` függvény segítségével nem csak konstans szövegeket, de matematikai kifejezéseket is lehetőségünk van megjeleníteni. Ilyenkor a fordító kiértékeli a kifejezést, kiszámítja annak eredményét, majd az eredményt fogja a ```print``` függvény a kimeneten megjeleníteni.

A következő 7 alapvető aritmetikai műveletet használhatjuk Python nyelven:

* Összeadás: ```+```

In [5]:
print(2+3)

5


 - Kivonás: ```-```

In [6]:
print(2-3)

-1


 - Szorzás: ```*```

In [7]:
print(2*3)

6


 - Osztás: ```/```

In [8]:
print(2/3)

0.6666666666666666


 - Maradékos osztás: ```%```

Az osztás maradékát fogja megadni. A példában: kettőben a három megvan nulla egésszer, és maradék a kettő.

In [9]:
print(2%3)

2


 - Hatványozás: ```**```

Az ```**``` operátor elé kerül az alap, utána pedig a kitevő. A példában: kettő a harmadik hatványon.

In [10]:
print(2**3)

8


- Egészosztás: ```//```

Megadja lefelé kerekítve az osztás eredményét. A példában: a három a kettőben nulla egésszer van meg.

In [11]:
print(2//3)

0


Lehetőségünk van összetettebb kifejezések megfogalmazására is. A végrehajtás tekintetében a matematikában megszokott szabályok érvényesülnek.

In [12]:
print(2*3-1)

5


In [13]:
print(3**3*3)

81


Zárójelek segítségével tudjuk szabályozni a végrehajtás sorrendjét.

In [14]:
print(2*(3-1))

4


In [15]:
print(3**(3*3))

19683


## Összetett üzenetek kiírása

Lehetőségünk van arra is, hogy a ```print``` segítségével összetett, több üzenetből álló kiírásokat hajtsunk végre. Például szeretnénk megjeleníteni egy szöveget, majd utána egy számítási eredményt is.

In [16]:
print("A kettő és a három szorzatának eredménye:", 2*3)

A kettő és a három szorzatának eredménye: 6


Ilyenkor az egyes üzeneteket vesszővel elválasztva soroljuk fel a zárójelen belül. Az előző példában először egy szöveget, majd egy matematikai kifejezés eredményét jelenítjük meg.

Természetesen kettőnél több paraméter is állhat itt, mindig vesszővel elválasztva egymástól.

In [17]:
print("Ada Lovelace 1815-ben született,", 2023-1815, "évvel ezelőtt. 1852-ben (", 1852-1815, ". évében) hunyt el.")

Ada Lovelace 1815-ben született, 208 évvel ezelőtt. 1852-ben ( 37 . évében) hunyt el.


Látható, hogy a vesszővel elválasztott elemek közé automatikusan egy-egy szóköz kerül a kimeneten. Ezért van az, hogy a ```2023-1815``` kifejezés előtti szövegrész végére és utáni szövegrész elejére nem raktunk szóköz karaktert. Viszont az ```1852-1815``` kifejezés esetén más a helyzet. Itt is láthatjuk az automatikus szóközeket, viszont itt egyik oldalra sem kellene.

Hogyan lehet ezt orvosolni?

A ```print``` függvény rendelkezik több opcionális paraméterrel. Az egyik ilyen a ```sep``` nevezetű, amely a "separator" szó rövidítéséből ered. Ennek segítségével megadható, hogy szóköz helyett milyen karaktert (vagy karaktersorozatot) illesszen be az egyes kiírandó elemek közé. Például megadhatjuk azt, a fenti szöveg esetén, hogy ne szerepeljen semmilyen elválasztó szöveg. Ehhez üres karaktersorozatot (```""```) kell értékül adni a ```sep``` paraméternek a következő módon:

In [18]:
print("Ada Lovelace 1815-ben született,", 2023-1815, "évvel ezelőtt. 1852-ben (", 1852-1815, ". évében) hunyt el.", sep="")

Ada Lovelace 1815-ben született,208évvel ezelőtt. 1852-ben (37. évében) hunyt el.


Látható, hogy az elválasztó szóközök eltűntek. Ezzel a zárójelbe írt életkor problémája megoldódott. Viszont a "208" előtt és után sincs így már szóköz, pedig ott szükség van rá. Ezért egyszerűbb, ha az előtte és utána található szövegrész végére és elejére fűzzük hozzá a kívánt elválasztó szóközöket. Így már megfelelő formában kapjuk meg a szöveget:

In [19]:
print("Ada Lovelace 1815-ben született, ", 2023-1815, " évvel ezelőtt. 1852-ben (", 1852-1815, ". évében) hunyt el.", sep="")

Ada Lovelace 1815-ben született, 208 évvel ezelőtt. 1852-ben (37. évében) hunyt el.


A ```sep``` paraméter értéke bármilyen más tetszőleges szöveg lehet, amit az adott feladat, elérendő cél megkíván. Például kifejezések eredményét pontosvesszővel tagoltan szeretnénk megjeleníteni. Ezt a következő módon tehetjük:

In [20]:
print(2**0, 2**1, 2**2, 2**3, 2**4, 2**5, 2**6, 2**7, sep=";")

1;2;4;8;16;32;64;128


A ```print``` függvény másik opcionális paramétere az ```end```. Ennek is egy karaktert vagy karaktersorozatot adhatunk meg. Ami az ```end``` értéke, azt a szöveget fogja az adott kiírás végére elhelyezni a ```print``` függvény. Ahogy a ```sep``` alapértelmezett értéke a szóköz, úgy az ```end``` alapértelmezett értéke a sortörés karakter. Ennek köszönhető, hogy ha egymás után több ```print``` függvényt hívunk meg, akkor az egyes kiírások egymás alatt, és nem egymás mellett fognak megjelenni a kimeneten.

Az előző példa alapján, ha az utolsó kifejezés után is szeretnénk pontosvesszőt megjeleníteni, akkor azt a következő módon tehetjük meg:

In [21]:
print(2**0, 2**1, 2**2, 2**3, 2**4, 2**5, 2**6, 2**7, sep=";", end=";")

1;2;4;8;16;32;64;128;

## Változók, típusok

A változó egy egyedi névvel rendelkező memóriaterület, ahol tetszőleges értéket tárolhatunk és felhasználhatjuk azt. Az értéknek mindig meghatározott típusa van (pl. egész szám, szöveg). A változók bármilyen érték tárolására képeseket, de egy időben mindig egy értéket tudnak tárolni.

### Int és float típusok, értékadás

Például létrehozhatunk egy változót azért, hogy egy számot tároljunk benne:

In [22]:
a = 42

Most létezik egy ```a``` nevű változó, aminek az értéke 42. Ezt most már fel tudjuk használni. Például kiírhatjuk az értékét a ```print``` függvény segítségével.

In [23]:
print(a)

42


De adhatunk hozzá tájékoztató szöveget is:

In [24]:
print("Az a nevű változó értéke:", a)

Az a nevű változó értéke: 42


Egy változó értékét bármikor megváltoztathatjuk. Mindig az utoljára elhelyezett érték marad érvényben, a korábbi elvész.

In [25]:
print(a)
a = 9
print(a)

42
9


Fontos, hogy csak olyan változót használhatunk fel, ami már létezik. Tehát legalább egyszer már adtunk neki értéket.

In [26]:
print(b)

NameError: name 'b' is not defined

A ```b``` változó értékét nem tudjuk kiírnik, hiszen az jelenleg nem létezik.

Egy változónak nem csak konstans számértéket adhatunk. Lehet az értéke matematikai kifejezés is. Itt ugyanaz van érvényben, mint a ```print``` függvény esetén. A kifejezés kiértékelésre kerül, majd annak az eredménye kerül a változóban eltárolásra.

In [27]:
b = 9 * 8
print(b)

72


Azt is megtehetjük, hogy egy változónak egy másik változó értékét adjuk át.

In [28]:
b = a
print(a)
print(b)

9
9


Ilyenkor, ha az ```a``` értékét változtatjuk, akkor a ```b``` értéke továbbra is az marad, ami legutoljára belekerült.

In [29]:
a = 2
print(a)
print(b)

2
9


Akár egészen összetett kifejezéseket is megadhatunk, amiben változó(k) is szerepel(nek).

In [30]:
b = a/10-2**a
print(b)

-3.8


Sőt, az is megengedett, hogy önmaga szerepeljen a kifejezésben. Ilyenkor a legutoljára elhelyezett érték kerül felhasználásra a képletben.

In [31]:
b = b * 2
print(b)

-7.6


A két utóbbi példából látható, hogy változóban nem csak egész számot, hanem valós számokat is el tudunk tárolni és műveletek tudunk rajtuk végezni. Az ilyen értékek tárolását a [lebegőpontos számábrázolás](https://en.wikipedia.org/wiki/Floating-point_arithmetic) teszi lehetővé. A Python nyelv, sok más nyelvhez hasonlóan, az [IEEE 754](https://en.wikipedia.org/wiki/IEEE_754)-es szabványát használja ehhez.

Ügyeljünk rá, hogy Python nyelven, mint a legtöbb programozás nyelv esetén is, tizedespontot használunk az égész rész és a tizedes törtek elválasztására. Például:

In [32]:
c = 0.5

Lehetőségünk van arra is, hogy egyszerre (egy sorban) több változónak is értéket adjunk. Ilyenkor az egyenlőség operátor bal oldalára vesszővel elválasztva soroljuk fel a változókat, amelyeknek értéket szeretnénk adni. Majd az egyenlőség jel jobb oldalán, szintén vesszővel elválasztva adjuk meg a kívánt értékeket. Olyan módon, hogy a sorban balról az első érték fog a sorban balról az első változóba kerülni. A második érték kerül a második változóba, és így tovább.

Ennek a lehetőségnek később több hasznát is fogjuk  venni. Most inkább csak jegyezzük meg, hogy van ilyen, és értsük meg a működését.

In [33]:
x, y, z = 1, 3, 5
print(x)
print(y)
print(z)

1
3
5


### Type függvény

A ```type``` függvény segítségével egy változóról megtudhatjuk, hogy az éppen benne szereplő érték milyen típust képvisel. Eddig 3 fajta típusú értékekkel találkoztunk: szöveggel, egész számokkal és valós számokkal.

Az ```a``` változó értéke jelenleg 2. Azaz típusa egész szám. Ezt láthatjuk ha lekérdezzük a ```type``` függvénnyel a típusát és azt rögtön ki is íratjuk a ```print``` függvény segítségével.

In [34]:
print(a)
print(type(a))

2
<class 'int'>


A ```<class 'int'>``` szöveg arra utal, hogy az ```a``` változó az ```int``` osztályba tartozik, azaz a jelenlegi értéke egy egész szám.

A ```b``` változónak jelenleg -7,6 az értéke, ami egy valós szám. Ezt látható, ha lekérdezzük az ő esetében is a típusát.

In [35]:
print(b)
print(type(b))

-7.6
<class 'float'>


Hasonló módon a ```c``` változó is ```float``` típusú, azaz valós szám lesz, hiszen a 0,5 értéket adtuk neki legutoljára.

In [36]:
print(c)
print(type(c))

0.5
<class 'float'>


A változó típusa mindig a benne található értéktől függ. Például az ```a``` változó értéke 2, ezért ő ```int``` típusú. De ha megváltoztatjuk 0,1-re az értékét, akkor már ```float``` lesz a típusa.

In [37]:
print(a)
print(type(a))
a = 0.1
print(a)
print(type(a))

2
<class 'int'>
0.1
<class 'float'>


### String típus használata

Változóknak tetszőleges szöveget is adhatunk értékül. Látható, hogy a típusa ```str``` lesz, ami a string szó rövidítéséből ered.

In [38]:
a = "Helló"
print(a)
print(type(a))

Helló
<class 'str'>


Ha ```a``` és ```b``` változók számok, akkor tudjuk az összegüket képezni ```c``` változóba. Ez mostanra már egyértelmű.

In [39]:
a = 2
b = 3
c = a + b
print(a, b ,c)

2 3 5


De mi a helyzet akkor, ha ```a``` és ```b``` szöveget tartalmaz? Nézzük meg!

In [40]:
a = "Helló,"
print(a)
b = " hogy vagy?"
print(b)
c = a + b
print(c)

Helló,
 hogy vagy?
Helló, hogy vagy?


Szövegek (```str```) esetén értelmezett a összeadás művelete, ami az úgynevezett konkatenációt fogja végrehajtani, azaz a két szöveget összefűzi.

Persze, mindez szám és szöveg között nem fog működni. Látható a lenti hibaüzenetből, hogy az ```int``` és a ```str``` típusok között az összeadás művelete nem értelmezett.

In [41]:
a = 3
b = "a"
c = a + b
print(c)

TypeError: unsupported operand type(s) for +: 'int' and 'str'

Ellenben a szorzást tudjuk értelmezni, mivel azt vissza tudjuk vezetni szövegek összeadására. Az összeadás pedig értelmezett, ezért a szorzás is. A példában a ```3 * "a"``` gyakorlatilag azt jelenti, mintha azt mondanánk, hogy ```"a" + "a" + "a"```. Azaz háromszor konkatenálja egymáshoz a megadott szöveget.

In [42]:
a = 3
b = "a"
c = a * b
print(c)

aaa


### Típus konverzió

Mi történik akkor, ha két különböző típusú számból szeretnénk egyet képezni? Például összeadunk egy egész és egy valós számot. Nézzük meg!

In [43]:
a = 0.1
b = 1
c = a + b
print(a, b, c)
print(type(a), type(b), type(c))

0.1 1 1.1
<class 'float'> <class 'int'> <class 'float'>


Látható, hogy az összeadás eredményeként a ```c``` változóba ```float``` érték került. Ilyenkor mindig a bővebb, nagyobb tartományt lefedő típus irányába történik a konverzió. Jelen esetben a 0,1 és az 1 összege 1,1 lett. Ezt ```int``` típusban nem tudjuk tárolni, legfeljebb csak az 1 értéket. Ezért a ```c``` változó ```float``` típusú lett. Ezt nevezzük implicit típus konverziónak.

A típus konverzió másik módja, ha szándékosan, "kézzel" jelezzük, hogy konverziót szeretnénk végrehajtani. Ilyen lehet például, ha az előző esetén maradva azt szeretnénk mondani, hogy az eredményül kapott 1,1 értékből mégis csak ```int``` típust csináljon, a pontosság vesztés ellenére is. Ehhez az ```int``` függvényt tudjuk használni, ami a paraméter kapott értékből egész számot fog készíteni.

In [44]:
a = 0.1
b = 1
c = int(a + b)
print(a, b, c)
print(type(a), type(b), type(c))

0.1 1 1
<class 'float'> <class 'int'> <class 'int'>


Hasonló módon használhatjuk a ```float``` függvényt is.

In [45]:
a = 3
b = float(a)
print(a, b)
print(type(a), type(b))

3 3.0
<class 'int'> <class 'float'>


Illetve létezik az ```str``` függvény is, amivel számból tudunk szöveget előállítani.

In [46]:
a = 42
b = str(a)
print(a)
print(b)
print(a*10)
print(b*10)

42
42
420
42424242424242424242


Az ```a``` változóban ```int``` típust tárolunk. Ennek az értékéből csinálunk ```str``` típust és ez kerül a ```b``` változóba. "Ránézésre" a kettő ugyanúgy jelenik meg. Mindkét esetben megjelenik a kimeneten a 42-es érték. De az egyik esetben, mint egész szám, a másik esetben pedig mint szöveg. A különbség jól látszik, hogy ha a szorzás műveletet alkalmazzuk. Az ```a``` változó esetén két egész számot, a 42-t és a 10-t szorozzuk össze a matimatikában megszokott szabályok szerint. A ```b``` változó esetén azonban konkatenációt hajtunk végre. Azaz a "42" szöveget 10-szer fűztük egymás után.

Az ```int``` és a ```float``` függvényeket arra is használhatjuk, hogy szövegként tárolt számot alakítsunk át számmá. Így lehetővé válik, hogy műveleteket hajtsunk végre rajta.

In [47]:
a = "42"
b = int(a)
c = float(a)
print(a, b, c)
print(type(a), type(b), type(c))

42 42 42.0
<class 'str'> <class 'int'> <class 'float'>


### Megjegyzések

A programkódban lehetőség van úgynevezett megjegyzések (comment) elhelyezésére. Ezek olyan részei a kódnak, melyek nem kerülnek végrehajtásra, ezért bármilyen tetszőleges szöveget tartalmazhatnak.

A megjegyzéseket két esetben használjuk:
 * Elsődlegesen segítségükkel a forráskódhoz magyarázó szövegeket tudunk hozzáfűzni. Ez által a kód átláthatóbb, könnyebben megérthető lesz.
 * Másodsorban ideiglenesen ki tudunk iktatni kódsorokat, ami a fejlesztés és tesztelés során sok esetben hasznos.

A megjegyzésnek két fajtáját használjuk Python nyelven:
  * egy soros megjegyzés és
  * több soros megjegyzés.

Az egy soros megjegyzés számjel (#) karakterrel kezdődik, amit tetszőleges szöveg követ:

In [48]:
# Ez itt egy egysoros megjegyzés. Bármit írhatok ide! ;-)

Általában úgy használjuk, hogy az értelmező, magyarázó szöveggel kiegészítendő kódsor vagy kódrészlet felett helyezzük el:

In [49]:
# Ez a print függvény egy üdvözlő szöveget ír a képernyőre.
print("Szia!")

Szia!


De lehetőség van arra is, hogy közvetlen egy utasítás után helyezzük el:


In [50]:
year = 2023 # Az aktuális év tárolására szolgáló változó
print(year)

2023


Előfordulhat, hogy egy hosszas kódrészlet kifejtő szövege túl hosszú és szeretnénk több sorban megjeleníteni, tördelni. Erre is van lehetőségünk, ezt nevezzük többsoros megjegyzésnek. Több soros megjegyzést egyrészt tudunk csinálni az eddig megismert egysoros megjegyzés alapján is:

In [51]:
# Ha egymás alá több számjel karaktert írunk,
# akkor több sorban,
# rendezetten,
# megjegyzéseket tudunk elhelyezni.

A legtöbb programozási nyelv fenn tart egy külön jelölésmódot a többsoros megjegyzések esetére. Ilyenkor általában csak a megjegyzés szakasz legelején és legvégén kell jeleznünk, hogy hol kezdődik és hol végződik a megjegyzés. Tehát nem kell minden egyes sor elején feltüntetni azt.

Szigorúan véve Python nyelven nincs lehetőség ilyen módon többsoros megjegyzéseket elhelyezni. Azonban mégis létezik erre egy megoldás, ami hivatalosan nem minősül megjegyzésnek, de a nemzetközi közösség által is elfogadott és használt módja ez a többsoros megjegyzésnek. Ebben az esetben a megjegyzés három darab idézőjel (```"```) vagy három darab aposztróf (```'```) karakterrel kezdődik. Ez után tetszőleges mennyiségű sorban írhatunk megjegyzéseket. A megjegyzés szakasz végét szintén három idézőjel vagy három aposztróf zárja. Fontos, hogy a nyitó és a záró karakterhármas ugyanolyan karakter legyen. Ezt a következő módon néz ki:

In [52]:
"""
Ez a program kiszámítja két évszám között eltelt évek számát.
A számítás eredményét a képernyőre írja formázott szövegként.

A változók jelentése:
 y0: Az egyik évszám, ennek kell a korábbinak lennie.
 y1: A másik évszám, ami a későbbi.
"""
y0 = 1970
y1 = 2023
print(y0, "és", y1, "között", y1 - y0, "év telt el.")

1970 és 2023 között 53 év telt el.


De ez miért nem minősül megjegyzésnek? Ezzel a módszerrel tulajdonképpen nem megjegyzést írunk, hanem többsoros karakterláncot (string-et). Ezt a módszert tudjuk alkalmazni olyan helyeken is, ahol string-re van szükségünk, de azokat több sorba szeretnénk rendezni:

In [53]:
a = """
ez most egy
több soros
szöveg
"""
print(a)


ez most egy
több soros
szöveg



Nagyon ritka, hogy a fenti módon több soros szöveget akarunk változóba helyezni, vagy bármi mást csinálni vele. De fontos érteni, hogy a megjegyzésnek ez a módszere szigorúan véve nem megjegyzés, hanem egy többsoros szöveg.

De akkor mitől lesz ez mégis megjegyzés?

A kódban bárhol elhelyezhetünk karaktersorozatot úgy, hogy azt semmi nem kapja bemenetként, vagy értékként, csupán a karaktersorozatot tartalmazza a kód adott része. Ilyenkor a fordító ugyan feldolgozza, de mivel nincs teendő vele (pl. nem kell egy változóban eltárolni), így nem csinál vele semmit. Olyan, mintha figyelmen kívül hagyná. Ebből adódóan ezt a hagyományos, egysoros string-el is megtehetjük, azonban ennek használata nem elfogadott a közösség által:

In [54]:
"A pring függvény kiírja az üzenetet a képernyőre."
print("blabla")

blabla


Tehát a fenti példa ugyan működik, de nem elfogadott a használata. Kizárólag többsoros megjegyzés esetén, tripla idézőjellel/aposztróffal megengedett és elfogadott ennek a használata. A fenti példát számjellel oldjuk meg, a következő módon:

In [55]:
# A pring függvény kiírja az üzenetet a képernyőre.
print("blabla")

blabla
